### PR TITLE
NO_MERGE_GIT_ISSUES_:( added aware toggle to easydao

### PR DESCRIPTION
**Pull request by michaelconnor468**

### CODE REVIEW

File: src/foam/dao/EasyDAO.js [hunk 1]
```diff
@@ -566,35 +566,41 @@ return delegate;
       name: 'testData',
       generateJava: false
     },
+    {
+      documentation: 'Enables automated adding of aware doas to qualifying decorator chains',
```
Comment (TW80000) [inline src/foam/dao/EasyDAO.js]:
```suggestion
      documentation: 'Enables automated adding of property-related DAO decorators to qualifying decorator chains',
```

File: src/foam/dao/EasyDAO.js [hunk 2]
```diff
@@ -566,35 +566,41 @@ return delegate;
...
+      name: 'enableAware',
+      class: 'Boolean',
+      value: true
+    },
     {
       name: 'serviceProviderAware',
       class: 'Boolean',
-      javaFactory: 'return foam.nanos.auth.ServiceProviderAware.class.isAssignableFrom(getOf().getObjClass());'
+      javaFactory: 'return !getEnableAware() ? false : foam.nanos.auth.ServiceProviderAware.class.isAssignableFrom(getOf().getObjClass());'
```
Comment (TW80000) [inline src/foam/dao/EasyDAO.js]:
```suggestion
      javaFactory: 'return getEnableAware() && foam.nanos.auth.ServiceProviderAware.class.isAssignableFrom(getOf().getObjClass());'
```

is simpler. Please change the others as well.

File: src/foam/dao/EasyDAO.js [hunk 3]
```diff
@@ -566,35 +566,41 @@ return delegate;
       name: 'testData',
       generateJava: false
     },
+    {
+      documentation: 'Enables automated adding of aware doas to qualifying decorator chains',
+      name: 'enableAware',
```
Comment (TW80000) [inline src/foam/dao/EasyDAO.js]: This is close enough to `EnabledAware` to be confusing. I think we should try to come up with a better name. `enablePropertyDecorators` maybe?

**Review by TW80000 — CHANGES_REQUESTED** [review]

File: src/foam/dao/EasyDAO.js (hunk 3 above)
Comment (michaelconnor468) [reply src/foam/dao/EasyDAO.js]: yahh that's probably a good idea; I'm going to change it to enableAwareDecorators

**Review by jlhughes — APPROVED** [review]
Sure, this works.

**Pull request closed without merging**